[ci][orc8r][helm][gha] push helm charts to artifactory (#7054)
magma/magma: docs/readmes/orc8r/deploy_build.md

```

> ## Build and publish Helm charts
> *Only Choose one option below*
> ### First option: Publish to artifactory

Reviewer: ```suggestion
## Build and publish Helm charts

NOTE: only choose one of the below options

### Option 1: Publish to artifactory
```

Reviewer: Actually, these two sections are confusing, because actually step 1 is optional and step 2 is required. So reword this or something to make things clearer

> We'll build the Orchestrator Helm charts, as well as publish them to an artifactory.

Reviewer: Can this be any Helm chart repo, or does it need to be a [ChartMuseum](https://chartmuseum.com/) instance? If the latter, we should specify as much

Author: The scripts are working against an artifactory helm repo, it would not work with an helm ChartMuseum instance or other type.

Reviewer: So is this script just something we'll use, special-purpose, for our magmacore repo? Or will it be generalize-able for anyone with a similar artifactory?

Define some necessary variables

```bash
export HELM_CHART_ARTIFACTORY_URL=ARTIFACTORY_URL
export HELM_CHART_MUSEUM_REPO=REPO_NAME
export HELM_CHART_MUSEUM_USERNAME=ARTIFACTORY_USERNAME
export HELM_CHART_MUSEUM_TOKEN=ARTIFACTORY_ACCESS_TOKEN
```

Next we'll run the package script. This script will package and publish the
necessary helm charts to the artifactory. The script expects a deployment
type to be provided, which will determine which orc8r modules are deployed.

> ### Second option: Publish to a private Github repo

Reviewer: ```suggestion
### Option 2: Publish to a private Github repo
```

## Build and publish Helm charts

NOTE: only choose one of the below options and then run the script

### Option 1: Publish to artifactory

We'll build the Orchestrator Helm charts, as well as publish them to an artifactory.

Define some necessary variables

```bash
export HELM_CHART_ARTIFACTORY_URL=ARTIFACTORY_URL
export HELM_CHART_MUSEUM_REPO=REPO_NAME
export HELM_CHART_MUSEUM_USERNAME=ARTIFACTORY_USERNAME
export HELM_CHART_MUSEUM_TOKEN=ARTIFACTORY_ACCESS_TOKEN
```

Next we'll run the package script. This script will package and publish the
necessary helm charts to the artifactory. The script expects a deployment
type to be provided, which will determine which orc8r modules are deployed.

> 

Reviewer: No need for double-newline here

```

Next we'll run the package script. This script will package and publish the
necessary Helm charts to the `GITHUB_REPO`. The script expects a deployment
type to be provided, which will determine which orc8r modules are deployed.

> 

Reviewer: Here and below, no need for double-newline

```

Next we'll run the package script. This script will package and publish the
necessary Helm charts to the `GITHUB_REPO`. The script expects a deployment
type to be provided, which will determine which orc8r modules are deployed.

### Run the package script

> 

Reviewer: No need for double-newline here

Run the package script

```bash

> $ ${MAGMA_ROOT}/orc8r/tools/helm/package.sh -d fwa [-v 1.5] # or chosen deployment type

Reviewer: Partners copy+paste these commands, they will get confused by the `[-v 1.5]` syntax. Either remove it here, or add a commented-out version of the command below showing the option

Run the package script

```bash
$ ${MAGMA_ROOT}/orc8r/tools/helm/package.sh -d fwa # or chosen deployment type

...

Uploaded orc8r charts successfully.
```
You can add `-v` option to overwrite the versions of the chart.
```bash

> $ ${MAGMA_ROOT}/orc8r/tools/helm/package.sh -d fwa  -v 1.5

Reviewer: Here and elsewhere: don't include the leading `$` unless you also include the output of the script below it
```suggestion
${MAGMA_ROOT}/orc8r/tools/helm/package.sh -d fwa  -v 1.5
```
